homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,338 @@
+"""Config flow for ezviz."""
+import logging
+from socket import error as SocketConnectionRefusedError, gaierror
+
+from pyezviz import EzvizClient, PyEzvizError
+from pyezviz.test_cam_rtsp import AuthTestResultFailed, TestRTSPAuth
+import requests.exceptions
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import CONN_CLASS_CLOUD_POLL, ConfigFlow, OptionsFlow
+from homeassistant.const import (
+    CONF_CUSTOMIZE,
+    CONF_IP_ADDRESS,
+    CONF_PASSWORD,
+    CONF_TIMEOUT,
+    CONF_TYPE,
+    CONF_URL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+
+from .const import (  # pylint: disable=unused-import
+    ATTR_SERIAL,
+    ATTR_TYPE_CAMERA,
+    ATTR_TYPE_CLOUD,
+    CONF_FFMPEG_ARGUMENTS,
+    DATA_COORDINATOR,
+    DEFAULT_CAMERA_USERNAME,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    EU_URL,
+    RUSSIA_URL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EzvizConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ezviz."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_CLOUD_POLL
+
+    async def _validate_and_create_auth(self, data):
+        """Try to login to ezviz cloud account and create entry if successful."""
+        await self.async_set_unique_id(data[CONF_USERNAME])
+        self._abort_if_unique_id_configured()
+
+        client = EzvizClient(
+            data[CONF_USERNAME],
+            data[CONF_PASSWORD],
+            data.get(CONF_URL, EU_URL),
+            data.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+        )
+
+        # Attempts a login request.
+        try:
+            await self.hass.async_add_executor_job(client.login)
+
+        except PyEzvizError as err:
+            raise InvalidAuth from err
+
+        except requests.exceptions.ConnectionError as err:
+            raise WrongURL from err
+
+        except requests.exceptions.HTTPError as err:
+            raise InvalidHost from err
+
+        auth_data = {
+            CONF_USERNAME: data[CONF_USERNAME],
+            CONF_PASSWORD: data[CONF_PASSWORD],
+            CONF_URL: data.get(CONF_URL, EU_URL),
+            CONF_TYPE: ATTR_TYPE_CLOUD,
+        }
+
+        return self.async_create_entry(title=data[CONF_USERNAME], data=auth_data)
+
+    def _wake_camera(self, serial):
+        """Wake a hybernating camera."""
+        if self.hass.data.get(DOMAIN):
+            for item in self.hass.data.get(DOMAIN):
```

Comment:
We can't rely on the integration here. We need to create a new instance of the ezviz api.

Reply:
Done. A new instance of the pyezviz api is created and used in config_flow.